### PR TITLE
Pull Request for Testing TicTacToe - RSpec 

### PR DESCRIPTION
**Pull request by abouhid**
This pull request is necessary for completing the testing of the TicTacToe Project, as asked on the Microverse Program.
This testing was performed using RSpec 3.9 in which we compare the results of simulated outputs to our classes' outputs.
The tests check the game flow, its parameters, and the winning/drawing scenarios.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -17,7 +17,8 @@ Metrics/ClassLength:
 Metrics/BlockLength:
   ExcludedMethods: ['describe']
   Max: 30
-
+Metrics/CyclomaticComplexity:
+  Enabled: false
```
Comment (ikraamg) [inline .rubocop.yml]: Please undo these changes and use a disable and enable line comment such as # rubocop:disable Metrics/CyclomaticComplexity if there is an exception case to use it.

File: lib/board.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#require 'colorize'
+# require 'colorize'
```
Comment (ikraamg) [inline lib/board.rb]: Is necessary for colorize to work? If not, please remove it.

**Review by ikraamg — CHANGES_REQUESTED** [review]
Good day @abouhid and @gdumani :smile: 
### **Status: Almost there! Minor changes required** :recycle: 

Very well done, you have good good test coverage and your tests all pass :heavy_check_mark:

A couple of things need to be fixed before we can have this merged: 

- [x] Please make sure the GitHub actions linter check is working. 
- [x] Locally there are quite a few linter Rubocop errors I see, please check that as well.
- [x] Best practices require you to keep the Rubocop .yml in the original state. Please don't change the file, rather use inline disabling comments for exception cases.
- [x] Please add information regarding your tests in your readme as your repo code is updated.
- [x] Although I do see you have one test for the board#current method, please separate it into a separate describe block and add a negative test case for it as well.
Specs for each method should be wrapped in one describe block. As in the example here https://github.com/rubocop-hq/rspec-style-guide#empty-lines-around-examples .  (screenshot  from  Pathwright)
- [x] Please add a test for board#display:
All the public methods of the classes implemented in project OOP are tested ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/all_tic_tac_toe.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

If you have any questions, don't hesitate to contact me through slack: **_@ikraam_**

_**I wish you all the best!**_  :desktop_computer: 

_Project reviewed by Ikraam Ghoor (TSE)_

**Comment by gdumani** [comment]
Added all corrections. I checked and reinstall Rubocop. Locally it works perfectly but we don't see anything wrong ins gitflows configuration. It still doesn't; run in the pull request

**Comment by ikraamg** [comment]
Great work getting it all done!

I am not sure what could be the issue but please check if GitHub actions is enabled by following these instructions:
https://docs.github.com/en/github/administering-a-repository/disabling-or-limiting-github-actions-for-a-repository#managing-github-actions-permissions-for-your-repository.

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
Hello Team :wave: 

Great job on doing the changes! :+1: 
All tests are passing :green_heart:  :test_tube: 
Let's cover one more detail together...

**Readme**

- [x]  As the previous TSE said, please add an automated tests section to inform users **how can they run the automated tests on their computer**. Assume users are non-technical people. Try to be descriptive as much as possible.

**Tets**

- [x]  For the **current method** test, I see you have one "it" block which has more than one logical test phrase. You can also divide it into two "it" blocks to be able to quickly understand what is the cause in case of an error. This will give you more control over your methods. Please try to implement this too.

That's it, team. Thanks for your efforts.
With all my respect :pray:

**Review by kubilaycaglayan — APPROVED** [review]
Hello Team :wave: 
Status: Approved :green_circle: 

Great job on implementing the changes.
Thanks for your efforts and time.

Your project is approved but please consider my message below before you merge :arrow_down: 
Note: Please work on your readme file, one more time to make it look more professional. 
You can use backticks ` to create code lines.

Something like this would look more professional:
- run `gem install rspec`
- and then, to run the tests: `rspec`

With all my respect... :pray:

**Pull request merged**